### DISCUSSION TITLE
Add predicate method to tell if record exists

### DISCUSSION THREAD

**Discussion by hi-tech-jazz** [jeremyevans/sequel#2100] [General]
Hi @jeremyevans!

Thanks for a great library.

I often find myself using  `Sequel::Dataset#empty?` combined with the `!` operator when I want to know if at least one record fulfilling some given conditions exists, however what I would like to write is really the inverse of said predicate to avoid the negation. The obvious name for such a method would be `exists?` (with the benefit of matching ActiveRecord) but in the case of Sequel, the resemblance to `Sequel::Dataset#exists` and `Sequel::Model#exists?` might cause confusion. Before I ponder more on the tricky question of naming I want to ask: Would you be open for considering the addition of such a method or is it an optimization for readability that feels unneeded?

**Comment by jeremyevans** [comment]
I don't want to add a method for this.  The best method name I can think of is `not_empty?` or `non_empty?`, neither of which I like.  `exists?` doesn't make sense to me, because a dataset exists regardless of the number of rows in it.  For users that want a method for the inverse of `empty?`, I recommend they add one themselves:

```ruby
class Sequel::Dataset
  def exists?; !empty? end
end

class Sequel::Model
  def self.exists?; dataset.exists? end
end
```

**Reply by JonathanFrias** [reply]
What about any?

**Reply by jeremyevans** [reply]
`any?` is already defined via Enumerable and does work, though it is certainly suboptimal for large tables.  I suppose we could optimize the case where no arguments or block are provided to act as `!empty?`.

**Reply by eriklovmo** [reply]
I noticed that [the any_not_empty extension](https://github.com/jeremyevans/sequel/commit/a1a2e02bcf) is already available, maybe that is good enough.

**Reply by jeremyevans** [reply]
Agreed, no need to include this in core if an extension is already available.  That extension should be updated, because since Ruby 2.5, `any?` can take a pattern argument. I'll work on that shortly.

**Reply by jeremyevans** [reply]
Fixed in e7286176dec19ba3a4c191e8c300feb5f5015fa9
